modules/core/04-channel/keeper/keeper.go

```diff
@@ -653,8 +671,46 @@ func (k Keeper) HasPruningSequenceStart(ctx sdk.Context, portID, channelID strin
 	return store.Has(host.PruningSequenceStartKey(portID, channelID))
 }
 
-// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than or equal to the pruning sequence.
-// The number of packet acknowledgements pruned is equal to limit. Pruning only occurs after a channel has been upgraded.
-func (Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit, pruningSequence uint64) error {
-	return nil
+// PruneStalePacketState prunes packet acknowledgements and receipts that have a sequence number less than pruning sequence end.
+// The number of packet acks/receipts pruned is equal to limit. Pruning can only occur after a channel has been upgraded.
+//
+// Pruning sequence start keeps track of the packet ack/receipt that can be pruned next. When it reaches pruningSequenceEnd,
+// pruning is complete.
+func (k Keeper) PruneStalePacketState(ctx sdk.Context, portID, channelID string, limit uint64) (uint64, uint64, error) {
+	if !k.HasPruningSequenceStart(ctx, portID, channelID) {
+		return 0, 0, errorsmod.Wrapf(types.ErrPruningSequenceStartNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+	pruningSequenceStart := k.GetPruningSequenceStart(ctx, portID, channelID)
+
+	pruningSequenceEnd, found := k.GetPruningSequenceEnd(ctx, portID, channelID)
+	if !found {
+		return 0, 0, errorsmod.Wrapf(types.ErrPruningSequenceEndNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	start := pruningSequenceStart
+	end := pruningSequenceStart + limit
+	for ; start < end; start++ {
+		// Stop pruning if pruningSequenceStart has reached pruningSequenceEnd, pruningSequenceEnd is
+		// set to be equal to the _next_ sequence to be sent by the counterparty.
+		if start >= pruningSequenceEnd {
+			break
+		}
+
+		if k.HasPacketAcknowledgement(ctx, portID, channelID, start) {
+			k.deletePacketAcknowledgement(ctx, portID, channelID, start)
```

Comment:
Nit, you can just delete here. Delete should be a no-op if it doesn't exist so you don't have to do the prior check has

Reply:
Changing this now.